internal/vision/provider.go

```
@@ -0,0 +1,274 @@
+		return Action{}, "", fmt.Errorf("vision: build request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	if p.APIKey != "" {
+		req.Header.Set("Authorization", "Bearer "+p.APIKey)
+	}
+
+	resp, err := p.Client.Do(req)
+	if err != nil {
+		return Action{}, "", fmt.Errorf("vision: chat request: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)